File: php/php-cli-skeleton/number.php
#!/usr/bin/env php
<?php
function main($argv) {       
	$opts = getopt(null, array('min:', 'max:'));
	
	$min = (int) isset($opts['min']) ? trim($opts['min']) : 0;
	$max = (int) isset($opts['max']) ? trim($opts['max']) : 100;
	
	puts("Guess The Number \n", 'green');
	puts("I'm thinking of a number between $min and $max \n\n");

	$tries = 0;
	$number = rand($min, $max);
	while (true) {
		if ($tries > 0) {
			puts('Try again: ');
		} else {
			puts('Take a wild guess: ');
		}
		
		$guess = trim(gets());
		$tries++;
		
		if (! is_numeric($guess)) {
			puts("It helps if you enter a number. \n", 'red');
		} elseif($guess < $min) {
			puts("Well it can't be less than $min can it. \n", 'red');
		} elseif($guess > $max) {
			puts("Well it can't be more than $max can it. \n", 'red');
		} elseif ($guess < $number) {
			puts("Try going higher. \n", 'red');
		} elseif ($guess > $number) {
			puts("Try going lower. \n", 'red');
		} else {
			puts("Correct, it was $number! \n", 'yellow');
			puts("It took you $tries guesses. \n", 'yellow');
			exit(0);
		}
	}
}

function gets() {
	return fgets(STDIN);
}

function puts($text, $style = null) {
	$styles = array(	
		'bold'		=> "\033[1m%s\033[0m",
		'black'		=> "\033[0;30m%s\033[0m",
		'red'		=> "\033[0;31;31m%s\033[0m",
		'green' 	=> "\033[0;32m%s\033[0m",
		'yellow'	=> "\033[0;33m%s\033[0m",
		'blue'		=> "\033[0;34m%s\033[0m",
		'magenta'	=> "\033[0;35m%s\033[0m",
		'cyan'		=> "\033[0;36m%s\033[0m",
	);
	
	fwrite(STDOUT, sprintf(isset($styles[$style]) ? $styles[$style] : "%s", $text));
}

if ('cli' === php_sapi_name() && basename(__FILE__) === basename($argv[0])) {
	main($argv);
}

File: python/python-cli-skeleton/number.py
#!/usr/bin/env python
import optparse, random, sys

def main():
	parser = optparse.OptionParser()
	parser.add_option('--min', type="int", default=0, help="The minimum number limit.")
	parser.add_option('--max', type="int", default=100, help="The maximum number limit.")
	opts, args = parser.parse_args()

	tries = 0
	number = random.randint(opts.min, opts.max)
		
	sys.stdout.write("Guess The Number \n")
	sys.stdout.write("I'm thinking of a number between %d and %d. \n\n" % (opts.min, opts.max))
		
	while True:
		if tries == 0:
			sys.stdout.write("Take a guess: ")
		else:
			sys.stdout.write("Try again: ")
		
		try:
			tries += 1
			guess = int(raw_input().strip())
		except ValueError:
			sys.stdout.write("It helps if you enter a number. \n")
			continue
		
		if guess < opts.min:
			sys.stdout.write("It can't be less than %d now can it! \n" % opts.min)
		elif guess > opts.max:
			sys.stdout.write("It can't be more than %d now can it! \n" % opts.max)
		elif guess < number:
			sys.stdout.write("Try going higher. \n")
		elif guess > number:
			sys.stdout.write("Try going lower. \n")
		else:
			sys.stdout.write("Correct, it was %d! \n" % number)
			sys.stdout.write("It took you %d guesses. \n" % tries)
			sys.exit(0)
		
if __name__ == '__main__':
	main()